### PR TITLE
React 18 support

### PR DESCRIPTION
**Pull request by DanishMehmood-bit**
**Changes include:**
- Added support for react@18, it is necessary for iTwinStudio upgrade to react 18.
- Removed support for react@16, I don't think anyone would be using react 16 anymore.
- Storybook@6 doesn't support react@18, I upgraded it regardless so that rush won't complain but it still uses react 17.
- Upgraded storybook to latest 6.x version.

**Tested the packages:**
- Storybook which uses react 17, its working fine.
- TestApp which uses react 18, its working fine.

**Future Plan:**
I'm planning on adding a react 18 based storybook probably storybook@8 but since its a rework from storybook@6 so I'll create the whole storybook from scratch and adding that in this PR will take alot of time and it'll halt the upgrade of iTwinStudio to react@18 as well.

### CODE REVIEW

**Review by raplemie — APPROVED** [review]
Changes seems reasonable, @aruniverse as previously discussed, could my name be removed from the codeowner and replaced with someone in platform, I'm getting further and further away from being able to properly make these reviews.

**Review by williamkbentley — APPROVED** [review]

File: packages/modules/imodel-browser/src/containers/iModelGrid/useIModelData.ts [hunk 1]
```diff
@@ -104,6 +102,7 @@ export const useIModelData = ({
     const paging = `&$skip=${skip}&$top=${top}`;
     const searching = searchText?.trim() ? `&$search=${searchText}` : "";
 
+    const abortController = new AbortController();
```
Comment (veekeys) [inline packages/modules/imodel-browser/src/containers/iModelGrid/useIModelData.ts:105]: Why did you need to change this?
I think now it might work a little different as we gonna recreate it everytime any of the deps change :?

File: packages/modules/manage-versions/src/components/ManageVersions/ManageVersions.tsx [hunk 2]
```diff
@@ -310,7 +310,9 @@ export const ManageVersions = (props: ManageVersionsProps) => {
               version: updatedVersion ? updatedVersion[0] : td.version,
             };
           });
-          setVersionsTableData(updatedVersionsTableData);
+          if (updatedVersionsTableData) {
```
Comment (veekeys) [inline packages/modules/manage-versions/src/components/ManageVersions/ManageVersions.tsx:313]: what for?

**Review by veekeys — COMMENTED** [review]
Have you checked if there were no warning of unsupported peer dep when you upgraded version to react 18?
Are you sure you want to remove react 16? I dont think it has any impact for us and you would be able to publish this as minor rather than major release.

**Comment by DanishMehmood-bit** [comment]
> Have you checked if there were no warning of unsupported peer dep when you upgraded version to react 18? Are you sure you want to remove react 16? I dont think it has any impact for us and you would be able to publish this as minor rather than major release.

Yes there aren't any warnings for unsupported peer dep.

Yes we should remove react 16 unless you know of any team who is still using react 16, we should keep our components up to date atleast keep them as close to core and core is already on react 18.

Well removing support for react 16 is a major release but it wouldn't impact any app not using react 16, but in general removing support for things which were previously supported comes under a major release

**Review by veekeys — APPROVED** [review]

**Comment by aruniverse** [comment]
> > Changes seems reasonable, @aruniverse as previously discussed, could my name be removed from the codeowner and replaced with someone in platform, I'm getting further and further away from being able to properly make these reviews.
> 
> Yes please replace all instances of @kckst8 & @raplemie in the codeowners with me in this PR

will this be done in a separate pr?

**Comment by DanishMehmood-bit** [comment]
> > > Changes seems reasonable, @aruniverse as previously discussed, could my name be removed from the codeowner and replaced with someone in platform, I'm getting further and further away from being able to properly make these reviews.
> > 
> > 
> > Yes please replace all instances of @kckst8 & @raplemie in the codeowners with me in this PR
> 
> will this be done in a separate pr?

@aruniverse Yes, this is the PR, please review that and then I'll merge that
https://github.com/iTwin/admin-components-react/pull/123
